src/Security/ContaoAuthenticator.php

```diff
@@ -28,6 +30,21 @@
 class ContaoAuthenticator implements SimplePreAuthenticatorInterface
 {
     /**
+     * @var ContainerInterface
+     */
+    private $container;
+
+    /**
+     * Constructor.
+     *
+     * @param ContainerInterface $container The service container
+     */
+    public function __construct(ContainerInterface $container)
```

Comment:
Currently it does load the user no matter the container scope. With the change, it would not, if the container is not injected (because that would be wrong).